test/inductor/test_compiled_autograd.py

```diff
@@ -2071,6 +2071,37 @@ def fn():
             sum(1 for e in expected_logs if e in logs.getvalue()), len(expected_logs)
         )
 
+    def test_verbose_logs_aot_id(self):
+        torch._logging.set_logs(compiled_autograd_verbose=True)
+
+        def fn():
+            model = torch.nn.Sequential(
+                torch.nn.Linear(4, 4),
+                torch.nn.ReLU(),
+                torch.nn.Linear(4, 4),
+                torch.nn.ReLU(),
+            )
+            x = torch.randn([2, 4])
+
+            @torch.compile
+            def forward(model, x):
+                return model(x)
+
+            result = forward(model, x).sum()
+            result.backward()
+            yield model[0].weight.grad
+            yield model[0].bias.grad
+            yield model[2].weight.grad
+            yield model[2].bias.grad
+
+        logs, ctx = logs_to_string(
+            torch._dynamo.compiled_autograd.__name__, "compiled_autograd_verbose"
```

Comment:
Should we always enable it instead of only under `compiled_autograd_verbose`? I believe this feature would be very useful for debugging internal jobs in the future, and if this aot id logging is on by default then we don't need to ask user to go back and run the job again with verbose enabled. (Or alternatively we can also ask user to always turn on `compiled_autograd_verbose`)

Reply:
yeah this one is not on the hot path so we can do that

Reply:
also just thinking:

(1) do compiled autograd logs gets turned on for most jobs internally?

(2) is there anything else we can put into tlparse from compiled autograd that we aren't right now?

Reply:
for 1, tlparse seems to have non-verbose logs by default

2, there's probably a bunch of cross-artifacts analysis we can do in tlparse. but i feel like adding logs to the compiled autograd artifacts is sufficient for the problems we're seeing rn

Reply:
@bdhirsh another issue we saw is that when going from CA Dynamo graph to AOTAutograd joint graph, the source of each node in AOTAutograd is lost (i.e. we don't know which Dynamo graph FX node it comes from anymore). Curious is there a way to preserve this information?

Reply:
hmm do you have an example, or a repro you can file?

I tried running the repro below, and when I the logs from `TORCH_LOGS="aot,+compiled_autograd"`, it seems like the stacktraces are lost when compiled autograd first gets a graph, before it sends it to AOTDispatcher. log output: P1528486054

code:
```
import torch


@torch.compile
def f(x):
    return torch.matmul(x, x).sin()

x = torch.randn(4, 4, requires_grad=True)
with torch._dynamo.utils.maybe_enable_compiled_autograd(True):
    out = f(x)
    out.sum().backward()

```

Reply:
need to use compiled_autograd_verbose instead of +compiled_autograd (it's a separate artifact since artifacts don't support verbosity levels)

Reply:
Filed here https://github.com/pytorch/pytorch/issues/133567, I'll take a look

Reply:
This seems to fix it https://github.com/pytorch/pytorch/pull/133574